### ISSUE TITLE
Formula parser broken for some variable names

### ISSUE THREAD

**Issue by kaskr** [glmmTMB/glmmTMB#1099]
Variable name 's' no longer works:

```r
library(glmmTMB)
s<-1:5
y<-rnorm(5)
glmmTMB(y~s)

Error in getParameterOrder(data, parameters, new.env(), DLL = DLL) : 
  Error when reading the variable: 'Z'. Please check data and parameters.
In addition: Warning messages:
1: In glmmTMB(y ~ s) : use of the ‘data’ argument is recommended
2: In getParameterOrder(data, parameters, new.env(), DLL = DLL) :
  Expected sparse matrix of class 'dgTMatrix'.
```

**Comment by bbolker** [comment]
ugh, I'll take a look. At least this is specific to `s`, not to the other "special" types (e.g. replacing `s` with `us` or `ar1` works OK)

**Comment by mebrooks** [comment]
Should I hold off on the submission to CRAN? 

The simple fix I tried broke a test in "smooth + diag() specials", so I'll leave it to you @bbolker.

**Comment by bbolker** [comment]
For what it's worth this is already present in the current CRAN version (it got in when we added processing for `s()` terms). It needs to be fixed in the `reformulas` package, now, so the good news is that it shouldn't hold up our submission anyway (although I will try to come up with a fix reasonably quickly - the proper way to do this is to write a recursive function that walks through the parse tree of the formula looking for specials, which I continue to find annoying to implement ...)

**Comment by bbolker** [comment]
This should be fixed [here](https://github.com/bbolker/reformulas/commit/c3ee5297f166da3d8f93bda150978ca0a55eda15) (not on CRAN yet).

**Comment by bbolker** [comment]
This is fixed with release 0.4.0 of `reformulas`
